modules/llrt_http/src/response.rs

```diff
@@ -141,34 +143,48 @@ impl<'js> Response<'js> {
         })
     }
 
+    async fn take_bytes_body<T>(&mut self, ctx: &Ctx<'js>, body: T) -> Result<Vec<u8>>
+    where
+        T: Body,
+        T::Error: std::fmt::Display,
+    {
+        let bytes = if let Some(abort_signal) = &self.abort_receiver {
+            select! {
+                err = abort_signal.recv() => return Err(ctx.throw(err)),
+                collected_body = body.collect() => collected_body.or_throw(ctx)?.to_bytes()
+            }
+        } else {
+            body.collect().await.or_throw(ctx)?.to_bytes()
+        };
+
+        if let Some(content_encoding) = &self.content_encoding {
+            let mut data: Vec<u8> = Vec::with_capacity(bytes.len());
+            match content_encoding.as_str() {
+                "zstd" => ZstdDecoder::new(&bytes[..])?.read_to_end(&mut data)?,
+                "br" => BrotliDecoder::new(&bytes[..]).read_to_end(&mut data)?,
+                "gzip" => GzDecoder::new(&bytes[..]).read_to_end(&mut data)?,
+                "deflate" => ZlibDecoder::new(&bytes[..]).read_to_end(&mut data)?,
+                _ => return Err(Exception::throw_message(ctx, "Unsupported encoding")),
+            };
+            Ok(data)
+        } else {
+            Ok(bytes.to_vec())
+        }
+    }
+
```

Comment:
Out of scope for this PR but we should be able to stream to the decompression decoders rather than from memory

Reply:
Agreed, also I think my cloned body might be short lived if we can land the streams.